src/main/daml/Daml/Finance/Interface/Types/Date/RollConvention.daml

```diff
@@ -22,6 +22,8 @@ data PeriodEnum
   -- ^ Week
   | Y
   -- ^ Year
+  | T
```

Comment:
I'm not sure I understand what this is supposed to mean ... could you give me an example?

Reply:
It basically means that there is no rolling schedule but just one period covering the whole term. It is example 7 here: https://www.fpml.org/spec/fpml-5-11-7-rec-1/html/confirmation/fpml-5-11-examples.html#s2

Reply:
I feel that `T` as an additional enum doesn't really give much information as to how long a period effectively is (which is what this type seems to be about). I feel we are adding an un-descriptive and ambiguous enumeration. 

How is the one period defined in the Fpml swap? 

Couldn't we as an alternative
- avoid rolling a schedule if there is no need to roll?
- write it as a schedule with an initial / final stub and zero regular periods?
- calculate the exact number of days from the Fpml swap representation, then use `D` as a period?

Reply:
> How is the one period defined in the Fpml swap?

```
<calculationPeriodFrequency>
<periodMultiplier>1</periodMultiplier>
<period>T</period>
<rollConvention>NONE</rollConvention>
</calculationPeriodFrequency>
```

Where the period enum contains a T in the schema:
https://www.fpml.org/spec/fpml-5-11-3-lcwd-1/html/confirmation/schemaDocumentation/schemas/fpml-shared-5-11_xsd/complexTypes/Frequency/period.html

Reply:
Thanks. What I meant to ask is when the one period starts and ends. 

From the code, it seems to start at `effectiveDate` and end at `terminationDate`. From the code, it seems that the schedule is not rolled at all in the case of a `T` period, which makes me think the additional enum is used only to deserialize an Fpml message.

If that's the case, I suggest duplicating `PeriodEnum` in `FpmlTypes.daml` and add the `T` enum there. This is because `T` has a well-defined meaning in the context of Fpml, but no well-defined meaning otherwise.

Reply:
Looking at the signature of this function, I think the logic could belong there:

```
-- | Generate schedule from a periodic schedule.
createSchedule : [HolidayCalendarData] -> PeriodicSchedule -> Schedule
createSchedule _ ps | Some errorMsg <- validDates ps = error errorMsg
createSchedule cals ps@PeriodicSchedule {businessDayAdjustment; effectiveDateBusinessDayAdjustment;
  terminationDateBusinessDayAdjustment, frequency, effectiveDate, terminationDate} =
    let
      (unadj, stubTypeInitial, stubTypeFinal) = case frequency.period of
        T -> ([effectiveDate, terminationDate], None, None)
        _ -> generateUnadjustedDatesAndStubInfo ps
      adj = adjust
        cals
        businessDayAdjustment
        effectiveDateBusinessDayAdjustment
        terminationDateBusinessDayAdjustment
        unadj
    in
      toSchedule unadj adj stubTypeInitial stubTypeFinal

```

Especially since the `adjust` and `toSchedule` part also apply to a single period `Schedule`.

Reply:
The schedule type is defined as:

```
-- | A schedule defined by a list of periods.
type Schedule = [SchedulePeriod]
```

and I think the case with only one period should ideally be covered by `createSchedule` and not be special cased with duplicate code.

Reply:
I have implemented this new `ScheduleFrequencyEnum` type now, to distinguish between schedules containing regular periods and a single term period.